BDMS-221-225: New measuring_point_history model

ksmuczynski opened the pull request:
<!--
- BDMS-221-225: New measuring_point_history model
-->
### **Why**
_This PR addresses the following problem / context:_
- This PR addresses the need to track a well's official measuring point height over time.
- The purpose of this PR is to make these intermediate updates available to @jacob-a-brown as we continue to co-work on the main BDMS-221 task.

### **How**
_Implementation summary - the following was changed / added / removed:_
- Created a new model, `measuring_point_history`, to store the official measuring point and description for a Thing.
- A new One-To-Many relationship named `measuring_points` was added to the `Thing` model. This addresses the fact that a Thing may have multiple measuring points over time.

### **Notes**
_Any special considerations, workarounds, or follow-up work to note?_
- This PR does NOT address updates to the schemas, tests, or transfer scripts.

ksmuczynski committed:
feat: create new `measuring_point_history` model.

The current schema lacks a way to track the authoritative measuring point height over time.

Created a new model, `measuring_point_history`, to store the official measuring point and description for a Thing. This table serves as a specialized historical log that tracks the measuring point over time.

ksmuczynski committed:
feat: add new relationship to `Thing` model.

A Thing may have multiple measuring points over time. As such, the `Thing` model requires a relationship to the new `measuring_point_history` model.

A new One-To-Many relationship named `measuring_points` was added to the `Thing` model.

ksmuczynski committed:
refactor: add new field to `measuring_point_history` model.

It would be useful to track the reason for updating the measuring point.

Create new `reason` field in the `measuring_point_history` model.

ksmuczynski committed:
Merge branch 'bdms-221' into kas-bdms-221-225-core-well-info-models-schemas

# Conflicts:
#	db/thing.py

jacob-a-brown commented on db/measuring_point_history.py:
```diff
@@ -0,0 +1,67 @@
+        nullable=False,
+        comment="The official, surveyed height of the measuring point relative to ground surface (in feet).",
+    )
+    mp_description: Mapped[str] = mapped_column(
```
can this be called `measuring_point_description` since it's called `measuring_point_height` above?

jacob-a-brown commented on db/measuring_point_history.py:
```diff
@@ -0,0 +1,67 @@
+
+    # --- Relationships ---
+    # Many-To-One: A description history record belongs to one Thing. Many history records may belong to a single Thing.
+    thing: Mapped["Thing"] = relationship("Thing", back_populates="mp_history")
```
`back_populates` should be `measuring_points` to correspond with how it is defined in the `thing` table.

jacob-a-brown commented on db/measuring_point_history.py:
since this model only pertains to the `thing` model I don't think that it needs to go into its own file. It can go into `db/thing.py`

ksmuczynski replied:
Happy to do this. Just curious, how are we deciding which models get their own files? When I'm looking for info about a table/model, I'm expecting it to be listed in the `db` directory.

jacob-a-brown replied:
I don't know if we have a standard, but when making a new model I put it into an existing file if it only pertains to other models in that file (e.g. `WellPurpose` and `WellCasingMaterial`). [Now that I think of it, I don't know if I've created a standalone model, but it makes sense for me to have the polymorphic models be in their own files for ease of understanding.]

ksmuczynski replied:
Curious what @jirhiker thoughts are.

jacob-a-brown replied:
@jirhiker I merged this so I can use develop off of it - where the table is defined in the codebase won't affect how it's used in the schemas/tests. If you think it should be in the `db/thing.py` file I can move it there easily.

ksmuczynski committed:
refactor: update 'measuring_point_history' model.

The `back_populates` parameter in the relationship section was updated to match the relationship name in the `THing` model.

The `mp_description` field was renamed `measuring_point_description` for clarity.
